### PR TITLE
[NU-1078] OIDC: Support for overrideFrontendAuthenticationStrategy configuration parameter

### PR DESCRIPTION
**Pull request by arkadius**
## Checklist before merge
- [x] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [x] Code is cleaned from temporary changes and commented out lines
- [x] Code formatting checked
- [x] Parts of the code that are not easy to understand are documented in the code
- [x] Changes are covered by automated tests
- [x] Showcase in dev-application.conf added to demonstrate the feature
- [x] Documentation added or updated
- [x] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [x] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [x] Verify that PR will be squashed during merge

### CODE REVIEW

File: security/src/main/scala/pl/touk/nussknacker/ui/security/oidc/OidcAuthenticationConfiguration.scala [hunk 1]
```diff
@@ -90,6 +95,7 @@ object OidcAuthenticationConfiguration {
   import net.ceedubs.ficus.readers.EnumerationReader._
   import pl.touk.nussknacker.engine.util.config.CustomFicusInstances._
   import pl.touk.nussknacker.ui.security.api.AuthenticationConfiguration.authenticationConfigPath
+  private implicit val valueReader: ValueReader[FrontendStrategySettings] = forDecoder
```
Comment (coutoPL) [inline security/src/main/scala/pl/touk/nussknacker/ui/security/oidc/OidcAuthenticationConfiguration.scala:98]: WDYT about moving the implicit val to the `create` method?

File: security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2AuthenticationResourcesSpec.scala [hunk 2]
```diff
@@ -102,4 +103,26 @@ class OAuth2AuthenticationResourcesSpec
       header[`Set-Cookie`] shouldBe Some(`Set-Cookie`(HttpCookie(name = cookieConfig.name, value = accessToken, httpOnly = true, secure = true, path = cookieConfig.path)))
     }
   }
+
+  it("should return authentication settings") {
+    Get(s"/authentication/oauth2/settings") ~> routes(authenticationResources()) ~> check {
+      status shouldBe StatusCodes.OK
+      val response = responseAs[Json]
+      response.hcursor.downField("authorizeUrl").as[String].value shouldBe defaultConfig.authorizeUri.toString + "?client_id=" + defaultConfig.clientId
```
Comment (coutoPL) [inline security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2AuthenticationResourcesSpec.scala:111]: maybe `response.hcursor.downField("authorizeUrl").as[String].value shouldBe s"${defaultConfig.authorizeUri.toString}?client_id=${defaultConfig.clientId}"` ?

File: security/src/main/scala/pl/touk/nussknacker/ui/security/oidc/OidcAuthenticationConfiguration.scala (hunk 1 above)
Comment (arkadius) [reply security/src/main/scala/pl/touk/nussknacker/ui/security/oidc/OidcAuthenticationConfiguration.scala:98]: And what about implicit imports? should be in the `create` as well? Looks like it is the only usage of them.

Comment (coutoPL) [reply security/src/main/scala/pl/touk/nussknacker/ui/security/oidc/OidcAuthenticationConfiguration.scala:98]: IMO it'd be nice to move them too. E.g. when we remove the `create` method in the future, it'd be easier to keep the code clean.

**Review by coutoPL — APPROVED** [review]
LGTM

**Pull request merged**